src/librustc/ty/layout.rs

```diff
@@ -546,7 +546,8 @@ pub struct Struct {
 
     pub min_size: Size,
 
-    /// Size of padding needed by the struct due to requested repr alignment
+    /// Size of padding needed by the struct due to requested repr alignment.
+    /// This is created as required and will be empty if the structs fields require no padding.
     pub padding: Vec<Size>,
```

Comment:
Sure, `[0 x i8]` or `{}` (the equivalent of our 0-tuple, `()`) both work to that effect (LLVM ZST).